build.gradle

```diff
@@ -1,67 +1,134 @@
+    // These can be tweaked, removed, or duplicated as needed.
+    runs {
+        client {
+            workingDirectory project.file('run')
+
+            // Recommended logging data for a userdev environment
+            property 'forge.logging.markers', 'SCAN,REGISTRIES,REGISTRYDUMP'
+
+            // Recommended logging level for the console
+            property 'forge.logging.console.level', 'debug'
+        }
 
-compileJava {
-    sourceCompatibility = targetCompatibility = "1.8"
-}
+        server {
 
-minecraft 
-{
-    version = "1.12.2-14.23.3.2655"
-    runDir = "run"
-    
-    mappings = "snapshot_20171003"
-    
-    clientJvmArgs = ["-Xmx1G"]
-    serverJvmArgs = ["-Xmx1G"]
-
-    replace "%VERSION%", project.version
-    replace "%MCLIB%", project.mclib
-    replace "%METAMORPH%", project.metamorph
-    replace "%BLOCKBUSTER%", project.blockbuster
+            // Recommended logging data for a userdev environment
+            property 'forge.logging.markers', 'SCAN,REGISTRIES,REGISTRYDUMP'
+
+            // Recommended logging level for the console
+            property 'forge.logging.console.level', 'debug'
+        }
+    }
 }
 
 dependencies {
-    compileOnly files("run/libs/blockbuster-${blockbuster}-${project.minecraft.version}-dev.jar")
-    compileOnly files("run/libs/metamorph-${metamorph}-${project.minecraft.version}-dev.jar")
-    compileOnly files("run/libs/mclib-${mclib}-${project.minecraft.version}-dev.jar")
+    // Specify the version of Minecraft to use, If this is any group other then 'net.minecraft' it is assumed
+    // that the dep is a ForgeGradle 'patcher' dependency. And it's patches will be applied.
+    // The userdev artifact is a special name and will get all sorts of transformations applied to it.
+    minecraft 'net.minecraftforge:forge:1.12.2-14.23.5.2860'
+
+    // You may put jars on which you depend on in ./libs or you may define them like so..
+    // compile "some.group:artifact:version:classifier"
+    // compile "some.group:artifact:version"
+
+    // Real examples
+    // compile 'com.mod-buildcraft:buildcraft:6.0.8:dev'  // adds buildcraft to the dev env
+    // compile 'com.googlecode.efficient-java-matrix-library:ejml:0.24' // adds ejml to the dev env
+
+    // The 'provided' configuration is for optional dependencies that exist at compile-time but might not at runtime.
+    // provided 'com.mod-buildcraft:buildcraft:6.0.8:dev'
+
+    // These dependencies get remapped to your current MCP mappings
+    // deobf 'com.mod-buildcraft:buildcraft:6.0.8:dev'
+
+    // For more info...
+    // http://www.gradle.org/docs/current/userguide/artifact_dependencies_tutorial.html
+    // http://www.gradle.org/docs/current/userguide/dependency_management.html
+
+    compileOnly files("run/libs/blockbuster-${blockbuster}-1.12.2-dev.jar")
+    compileOnly files("run/libs/metamorph-${metamorph}-1.12.2-dev.jar")
+    compileOnly files("run/libs/mclib-${mclib}-1.12.2-dev.jar")
```

Comment:
Use the ${project.minecraft.version} variable, there was no need to change this.